pkg/apis/clusterlink.net/v1alpha1/export.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Export defines a service being exported by the local Peer for use by others.
+// Only explicitly exported services can be accessed remotely.
+type Export struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the attributes of the exported service.
+	Spec ExportSpec `json:"spec,omitempty"`
+}
+
+// ExportSpec contains all attributes of an exported service.
+type ExportSpec struct {
+	// Port of the exported service.
```

Comment:
What about the service?
https://github.com/clusterlink-net/clusterlink/pull/287/files#:~:text=ExportSpec%20struct%20%7B-,Service%20string,-//%20name%20of

Reply:
For start I only implement support for service which equals the export name.
If we add support for custom service name, we can add that field then.